src/testrender/cuda/rend_lib.cu

```diff
@@ -20,56 +21,56 @@ OSL_NAMESPACE_EXIT
 extern "C" {
 
     __device__
-    void* closure_component_allot (void* pool, int id, size_t prim_size, const float3& w)
+    void* closure_component_allot (void* pool, int id, size_t prim_size, const OSL::Color3& w)
     {
-        ((ClosureComponent*) pool)->id = id;
-        ((ClosureComponent*) pool)->w  = w;
+        ((OSL::ClosureComponent*) pool)->id = id;
+        ((OSL::ClosureComponent*) pool)->w  = w;
 
-        size_t needed   = (sizeof(ClosureComponent) - sizeof(void*) + prim_size + 0x7) & ~0x7;
+        size_t needed   = (sizeof(OSL::ClosureComponent) - sizeof(void*) + prim_size + 0x7) & ~0x7;
         char*  char_ptr = (char*) pool;
 
         return (void*) &char_ptr[needed];
```

Comment:
Not really related to this review, but does anyone have any thoughts on how to handle overflow conditions? If I understand correctly, we are just hoping the pool is big enough here ...

Reply:
Sizing the closure pool is definitely a can of worms. Is it possible to statically bound the pool?

Reply:
I might be possible by just counting the closure creation/add/mul instructions in the post-optimized group -- but that is likely to be a very conservative value (ie: much bigger than necessary).

Its not an urgent issue -- just wondering what kind of error handling we might have available to us to at least diagnose when this happens.